File: Love-Babbar-450-In-CSharp/04_searching_and_sorting/04_square_root_of_an_integer.cs
using System;
using System.Collections.Generic;
using System.Text;
using Xunit;

namespace _04_searching_and_sorting
{
    public class _04_square_root_of_an_integer
    {
        [Fact]
        public void reverse_arrayTest()
        {


        }


        /*
            link: https://practice.geeksforgeeks.org/problems/count-squares3649/1
        */



        // ----------------------------------------------------------------------------------------------------------------------- //
        /*
            note: said "less than N"
            TC: O(1)
        */
        private double countSquares1(double N)
        {

            double ans = Math.Sqrt(N - 1);

            return ans;
        }



        // ----------------------------------------------------------------------------------------------------------------------- //
        /*
            TC: O(sqrt(n-1))
        */
        private int countSquares2(int N)
        {
            // code here
            int i = 1;
            int count = 0;
            while ((i * i) < N)
            {
                i++;
            }
            return i - 1;
        }

    }
}

File: Love-Babbar-450-In-CPPTest/04_searching_and_sorting/04_square_root_of_an_integer.cpp
/*
    link: https://practice.geeksforgeeks.org/problems/count-squares3649/1
*/

#include "CppUnitTest.h"
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;
using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace LoveBabbar450InCPPTest
{
    TEST_CLASS(SearchSort)
    {
    public:

        TEST_METHOD(find1Test)
        {
            std::string charM = "harhs";
            int age = 14;
            age = 55;
            std::string lastName = "<<charM <<singh";

        }
    };
}

// ----------------------------------------------------------------------------------------------------------------------- //
/*
    note: said "less than N"
    TC: O(1)
*/
int countSquares(int N) {

    int ans = sqrt(N - 1);

    return ans;
}



// ----------------------------------------------------------------------------------------------------------------------- //
/*
    TC: O(sqrt(n-1))
*/
int countSquares2(int N) {
        // code here
    int i = 1, count = 0;
    while ((i * i) < N) i++;
    return i - 1;
}
